src/dmd/cparse.d

```diff
@@ -3450,11 +3450,6 @@ final class CParser(AST) : Parser!AST
                 return false;
             t = peek(t);
         }
-        else if (declarator != DTR.xdirect)
-        {
-            pt = t;
-            return true;        // declarator is optional
-        }
```

Comment:
I'm hitting issues with this, what should have been done is the condition and body inversed.
```
else if (declarator != DTR.xabstract)
    return false;
```

Reply:
The best thing is to make a PR for that, with the test cases you discovered.

Reply:
I did, when dealing with the failing tests in #12724.